File: 0739-daily-temperatures/0739-daily-temperatures.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        deque<int> myDeque;
        myDeque.push_back(0);
        vector<int> res(temperatures.size(), 0);

        for (int i = 1; i < temperatures.size(); i++) {
            if (temperatures[i] <= temperatures[myDeque.back()]) {
                myDeque.push_back(i);
            } else {
                while(!myDeque.empty() && temperatures[myDeque.back()] < temperatures[i]) {
                    res[myDeque.back()] = (i - myDeque.back());
                    myDeque.pop_back();
                }
                myDeque.push_back(i);
            }
        }

        return res;
    }
};

File: 0739-daily-temperatures/0739-daily-temperatures.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        stack = [0]
        results = [0] * len(temperatures)

        for i in range(1, len(temperatures)):
            while stack and temperatures[i] > temperatures[stack[-1]]:
                day = stack.pop()
                results[day] = i - day
            stack.append(i)

        return results
        